### PR TITLE
Fee warning

### PR DESCRIPTION
**Pull request by Vitalsine85**
Added fee warnings to modals to improve visibility. Thanks Fabiaz

### CODE REVIEW

File: src/hooks/useBlockDiff.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    ethereum,
+  }: { account: string; ethereum: provider } = useWallet()
+  const block = useBlock()
+  const panda = usePanda()
+  const masterChefContract = getMasterChefContract(panda)
+  const firstDepositBlock = useFirstDepositBlock (pid)
+  const lastWithdrawBlock = useLastWithdrawBlock (pid)
+  
+  const blockDiff =
+  block -
```
Comment (jondwillis) [inline src/hooks/useBlockDiff.ts:27]: formatting seems odd here, double check it

File: src/hooks/useBlockDiff.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import { useWallet } from '@binance-chain/bsc-use-wallet'
+import useBlock from './useBlock'
+import useFirstDepositBlock from './useFirstDepositBlock'
+import useLastDepositBlock from './useLastDepositBlock'
+import useLastWithdrawBlock from './useLastWithdrawBlock'
+import { getMasterChefContract } from '../panda/utils'
+import usePanda from './usePanda'
+import BigNumber from 'bignumber.js'
+
+// import debounce from 'debounce'
```
Comment (jondwillis) [inline src/hooks/useBlockDiff.ts:13]: rm

File: src/panda/utils.ts [hunk 3]
```diff
@@ -243,6 +243,51 @@ export const getStaked = async (
...
+  pid: number,
+  account: string,
+): Promise<BigNumber> => {
+  try {
+    const { lastDepositBlock } = await masterChefContract.methods
+      .userInfo(pid, account)
+      .call()
+    return new BigNumber(lastDepositBlock)
+  } catch {
+    return new BigNumber(0)
```
Comment (jondwillis) [inline src/panda/utils.ts:287]: it might be better to return null or undefined

File: src/views/Farm/components/Fee.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+	color: ${(props) => props.theme.color.grey[400]};
+	font-size: 16px;
+	font-weight: 400;
+	margin: 0;
+	padding: 0;
+	text-align: center;
+	max-width: 750px;
+`
+const Warning = styled.h3`
+color: red;
```
Comment (jondwillis) [inline src/views/Farm/components/Fee.tsx:83]: formatting looks off in this file-- maybe it is just GitHub

File: src/views/Farm/components/FeeWarning.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+const FeeWarning: React.FC = () => {
+    return (
+            <StyledDocsWarning>
+            <p>First Deposit Block:</p>
+            <p>Last Withdraw Block:</p>
+            <p>Current Block:</p>
+            <p>Withdraw Fee:</p>
+            </StyledDocsWarning>
+		)
```
Comment (jondwillis) [inline src/views/Farm/components/FeeWarning.tsx:30]: formatting looks off

File: src/views/Farm/components/WithdrawModal.tsx [hunk 6]
```diff
@@ -7,18 +7,21 @@ import ModalTitle from '../../../components/ModalTitle'
...
 	tokenName?: string
+	pid: number
 }
 
 const WithdrawModal: React.FC<WithdrawModalProps> = ({
 	onConfirm,
 	onDismiss,
 	max,
 	tokenName = '',
+	pid = 0
```
Comment (jondwillis) [inline src/views/Farm/components/WithdrawModal.tsx:24]: maybe better to allow undefined or null and just not show the warning

File: tsconfig.json [hunk 7]
```diff
@@ -18,7 +18,8 @@
     "isolatedModules": true,
     "noEmit": true,
     "jsx": "react",
-    "strictNullChecks": false
+    "strictNullChecks": false,
+    "noImplicitAny": false
```
Comment (jondwillis) [inline tsconfig.json:22]: we probably want this-- did you specifically do it to avoid having to typescript-ify "the edges"?

**Review by jondwillis — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 7 above)
Comment (rocky-baoboa) [reply tsconfig.json:22]: `noImplicitAny` shouldn't be set, instead you can simply explicitly set something to `any`. 

IE: 

```
function name(arg1: number, arg2) {
  return 1;
}
```

Would throw the `noImplicitAny` error, but you can just do this

```
function name(arg1: number, arg2: any): any {
  return 1;
}
```

Comment (Vitalsine85) [reply tsconfig.json:22]: thanks for the walkthrough @rocky-baoboa !

**Review by jondwillis — APPROVED** [review]

**Pull request merged**